File: DynamicKey/AgoraDynamicKey/nodejs/sample/ChatTokenBuilderSample.js
const ChatTokenBuilder = require("../src/ChatTokenBuilder").ChatTokenBuilder;

// Need to set environment variable AGORA_APP_ID
const appId = process.env.AGORA_APP_ID;
// Need to set environment variable AGORA_APP_CERTIFICATE
const appCertificate = process.env.AGORA_APP_CERTIFICATE;

const userUuid = "a7180cb0-1d4a-11ed-9210-89ff47c9da5e";
const expirationInSeconds = 600;

console.log("App Id:", appId);
console.log("App Certificate:", appCertificate);
if (appId == undefined || appId == "" || appCertificate == undefined || appCertificate == "") {
    console.log("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE");
    process.exit(1);
}

const userToken = ChatTokenBuilder.buildUserToken(appId, appCertificate, userUuid, expirationInSeconds);
console.log("Chat User Token:", userToken);

const appToken = ChatTokenBuilder.buildAppToken(appId, appCertificate, expirationInSeconds);
console.log("Chat App Token:", appToken);

File: DynamicKey/AgoraDynamicKey/ruby/sample/chat_token_builder_sample.rb
require_relative '../lib/dynamic_key2'

# Need to set environment variable AGORA_APP_ID
app_id = ENV['AGORA_APP_ID']
# Need to set environment variable AGORA_APP_CERTIFICATE
app_certificate = ENV['AGORA_APP_CERTIFICATE']

user_id = '2882341273'
token_expiration_in_seconds = 600

puts "App Id: #{app_id}"
puts "App Certificate: #{app_certificate}"
if !app_id || app_id == '' || !app_certificate || app_certificate == ''
  puts 'Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE'
  exit
end

token = AgoraDynamicKey2::ChatTokenBuilder.build_user_token(app_id, app_certificate, user_id,
                                                            token_expiration_in_seconds)
puts "Chat user token : #{token}"

token = AgoraDynamicKey2::ChatTokenBuilder.build_app_token(app_id, app_certificate, token_expiration_in_seconds)
puts "Chat app token: #{token}"
